android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TabListReducer.kt

```diff
@@ -287,9 +287,6 @@ private fun findNewSelectedTabId(
         // We found a nearby tab, let's select it.
         nearbyTab != null -> nearbyTab.id
 
-        // If there's no private tab to select anymore then just select the last regular tab
-        isPrivate -> tabs.last().id
-
```

Comment:
This was causing [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1874078) and I cannot otherwise find any behavior where we are expecting a normal mode tab to be selected when all private tabs are closed.

Is this worth a changelog entry?

Reply:
Ya, let's add a note about this because it is noteworthy. I didn't have a chance to test this on AC sample-browser. Could you take a quick look over there and report if everything is working as expected?

Reply:
I'm actually not familiar with the expected behavior of sample-browser 😅 how do I add private tabs?

Reply:
It was entirely possible the sample-browser didn't handle adding private tabs or removal of all tabs. My general ask is to just open the sample-browser and create some tabs and remove some/all the tabs to continue to test these AC changes and ensure nothing is breaking.

Reply:
Yeah everything looks good there